pkg/vendors/mellanox/mellanox.go

```diff
@@ -0,0 +1,419 @@
+package mellanox
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	sriovnetworkv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/global/consts"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
+)
+
+// BlueField mode representation
+type BlueFieldMode int
+
+const (
+	BluefieldDpu BlueFieldMode = iota
+	BluefieldConnectXMode
+
+	internalCPUPageSupplier   = "INTERNAL_CPU_PAGE_SUPPLIER"
+	internalCPUEswitchManager = "INTERNAL_CPU_ESWITCH_MANAGER"
+	internalCPUIbVporto       = "INTERNAL_CPU_IB_VPORT0"
+	internalCPUOffloadEngine  = "INTERNAL_CPU_OFFLOAD_ENGINE"
+	internalCPUModel          = "INTERNAL_CPU_MODEL"
+
+	ecpf        = "ECPF"
+	extHostPf   = "EXT_HOST_PF"
+	embeddedCPU = "EMBEDDED_CPU"
+
+	disabled = "DISABLED"
+	enabled  = "ENABLED"
+
+	VendorMellanox = "15b3"
+	DeviceBF2      = "a2d6"
+	DeviceBF3      = "a2dc"
+
+	PreconfiguredLinkType = "Preconfigured"
+	UknownLinkType        = "Uknown"
+	TotalVfs              = "NUM_OF_VFS"
+	EnableSriov           = "SRIOV_EN"
+	LinkTypeP1            = "LINK_TYPE_P1"
+	LinkTypeP2            = "LINK_TYPE_P2"
+	MellanoxVendorID      = "15b3"
+)
+
+type MLXNic struct {
+	EnableSriov bool
+	TotalVfs    int
+	LinkTypeP1  string
+	LinkTypeP2  string
+}
+
+//go:generate ../../../bin/mockgen -destination mock/mock_mellanox.go -source mellanox.go
+type MellanoxInterface interface {
```

Comment:
I can't because they are all part of the same interface 

```
type HostHelpersInterface interface {
	utils.UtilsInterface
	host.HostManagerInterface
	host.StoreManagerInterface
	mlx.MellanoxInterface
}
```

Reply:
ack, interface embedding in struct will yield a "redeclaration" error.